### PR TITLE
get search-reindex-api client via api-router

### PR DESCRIPTION
**Pull request by justinpjose**
### What
**Describe what you have changed and why.**
- The app needs to talk to `search-reindex-api` via the `api-router`
- This has now been done and the unit test has been updated accordingly
- Component test was made to test the `healthcheck` endpoint of the app

### How to review
**Describe the steps required to test the changes.**
- Check if the code changes make sense
- Check if the unit test and component test passes

### Who can review
**Describe who worked on the changes, so that other people can review.**
- Anyone

### CODE REVIEW

File: features/steps/component.go [hunk 1]
```diff
@@ -57,35 +82,62 @@ func NewSearchReindexTrackerComponent() (*SearchReindexTrackerComponent, error)
...
+	componentBuildTime := strconv.Itoa(int(time.Now().Unix()))
+	versionInfo, err := healthcheck.NewVersionInfo(componentBuildTime, gitCommitHash, appVersion)
+	if err != nil {
+		return nil, err
+	}
+	hc := healthcheck.New(versionInfo, cfg.HealthCheckCriticalTimeout, cfg.HealthCheckInterval)
+	return &hc, nil
+}
+
+func (c *SearchReindexTrackerComponent) DoGetHealthClient(name string, url string) *health.Client {
```
Comment (RMPall) [inline features/steps/component.go]: (suggestion): I guess  its probably a good idea to check the name and url for empty string and return a nil client if it is.

Comment (RMPall) [inline features/steps/component.go]: (suggestion): perhaps refactor this as ``` DoGetHealthClient(name, url string)```

File: features/steps/events.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+	"github.com/rdumont/assistdog"
+)
+
+func theseEventsAreConsumed[M event.KafkaAvroModel](c *SearchReindexTrackerComponent, schema *avro.Schema, consumer kafka.IConsumerGroup, table *godog.Table) error {
+	observationEvents, err := convertToEvents[M](table)
+	if err != nil {
+		return fmt.Errorf("failed to convert to events - err: %v", err)
+	}
+
+	//signals := registerInterrupt()
```
Comment (RMPall) [inline features/steps/events.go]: (minor): can remove this if it's not being used instead of commenting out ?

File: features/steps/events.go [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+			return fmt.Errorf("failed to marshal events - err: %v", err)
+		}
+
+		consumer.Channels().Upstream <- kafkatest.NewMessage(bytes, 0)
+	}
+
+	time.Sleep(300 * time.Millisecond)
+
+	// kill application
+	//signals <- os.Interrupt
```
Comment (RMPall) [inline features/steps/events.go]: (minor): can remove this if it's not being used instead of commenting out ?

File: features/steps/helper.go [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+import (
+	"net/http"
+
+	"github.com/maxcnunes/httpfake"
+)
+
+func healthCheckStatusHandle(status int) httpfake.Responder {
+	return func(w http.ResponseWriter, r *http.Request, rh *httpfake.Request) {
+		rh.Lock()
```
Comment (RMPall) [inline features/steps/helper.go]: (question): why do you need this lock and unlock here i.e why should this be thread safe ? I don't think this is needed but may be i am missing something ?

File: features/steps/component.go (hunk 1 above)
Comment (justinpjose) [reply features/steps/component.go]: Done

Comment (justinpjose) [reply features/steps/component.go]: Done

File: features/steps/events.go (hunk 2 above)
Comment (justinpjose) [reply features/steps/events.go]: Done

File: features/steps/events.go (hunk 3 above)
Comment (justinpjose) [reply features/steps/events.go]: Done

File: features/steps/helper.go (hunk 4 above)
Comment (justinpjose) [reply features/steps/helper.go]: Done

**Review by RMPall — APPROVED** [review]

**Pull request merged**